ai_generation_functions.py

```diff
@@ -10,19 +12,49 @@
 from requests.exceptions import Timeout, RequestException
 import gradio as gr
 from prompts import TITLE_GENERATION_PROMPT, LYRICS_GENERATION_PROMPT, SONG_STYLE_GENERATION_PROMPT, LYRICS_ANALYSIS_SONG_STYLE_PROMPT, LYRICS_ANALYSIS_INSTRUMENTS_PROMPT, music_categories
-
+import tempfile
+import PyPDF2
 
 load_dotenv()
 client = OpenAI()
 
 @observe()
 def get_lyrics(youtube_link):
-    try:
-        with open('tests/ly_1.txt', 'r', encoding='utf-8') as f:
-            lyrics = f.read()
-        return lyrics if lyrics else "No lyrics found."
-    except Exception as e:
-        return f"An error occurred: {str(e)}"
+        video_id = extract.video_id(youtube_link)
+        # Try to get the transcript (subtitles) in English
+        try:
+            transcript_list = YouTubeTranscriptApi.list_transcripts(video_id)
+            transcript = transcript_list.find_transcript(['en'])
```

Comment:
One thing is that the song chosen by the user is very likely to be in their native language, so they would probably want to see the lyrics in their own language. 

If we use subtitles, the user might have to select the language manually. That's why I was thinking of just using Whisper instead.

Reply:
Another thing is, if you extract YouTube using subtitles and your MP3 extraction uses Whisper, both methods achieve the same result but may behave differently, which could confuse the user.

Reply:
in regards to youtube subtitles and whisper I thought it would be nicer initially to have a failsafe so that if one fails the other kicks in. On review trying to use pure whisper I am encoutring lots of errors which now explains why i used the 2. Maybe, I may not know how but I thought the 2 would be nice to have them all.